--- a/lib/Dialect/Triton/IR/Dialect.cpp
+++ b/lib/Dialect/Triton/IR/Dialect.cpp
@@ -40,8 +40,8 @@ void TritonInlinerInterface::handleTerminator(Operation *op,
     return;
 
   // Replace the return with a branch to the dest.
   OpBuilder builder(op);
-  mlir::cf::BranchOp::create(builder, op->getLoc(), newDest,
-                             returnOp.getOperands());
+  builder.create<mlir::cf::BranchOp>(op->getLoc(), newDest,
+                                     returnOp.getOperands());
   op->erase();
 }
 
